### DISCUSSION TITLE
I am struggling to redirect on self hosted

### DISCUSSION THREAD

**Discussion by audas** [FreshRSS/FreshRSS#7001] [General]
I am hosting on my own local network, static ip and have Apache2 with an httpd.conf file redirecting things.

So I have a remote url eg "labs.media.com/rss" which redirects to my static home server at 144.x.xx.xx this is picked up by Apache and is rerouting based on the /rss with some re-write conditions.

```
    # Redirect /freshrss to 192.168.20.35:8080
    RewriteCond %{REQUEST_URI} ^/rss
    RewriteRule ^/rss(.*)$ http://192.168.20.35:8080$1 [P,L]

    ProxyPass /rss http://192.168.20.35:8080/
    ProxyPassReverse /rss http://192.168.20.35:8080/`
```

I have tried all manner of conditions and re-writes to  .

The page re-writes to 144.x.xx.xx/i/?rid=6736e57c90d3c  URL Not Found 

BUT if I manually change it to 144.x.xx.xx/rss/i/?rid=6736e57c90d3c then it works.

The problem is I can not change the rewrite conditions in the apache config to include the /rss as this means the landing site is not found.

So I included an apache config inside the Docker as suggested and ensured this was included to change the root however this also did not work.

Very confused here is someone could provide some insight please?

```
Alias /rss /var/www/FreshRSS

<Directory /var/www/FreshRSS>
    Options FollowSymLinks
    AllowOverride All
    Require all granted
</Directory>
DocumentRoot /var/www/FreshRSS
```

**Comment by Alkarex** [comment]
Check https://github.com/FreshRSS/FreshRSS/blob/edge/Docker/README.md#alternative-reverse-proxy-using-apache

**Reply by audas** [reply]
Hi Alex - thanks for this reply. 
This is where I was at - the proxy redirects perfectly well. FreshRSS is running in a Docker port 8080

It lands on this Docker which then reloads to  "http://144.x.xx.xx/i/?rid=6737c381559ca" 

If I manually amend this url with "/rss" so it looks like this "http://144.x.xx.xx/rss/i/?rid=6737c381559ca" then it works. 

Is FreshRSS struggling because I have not yet implemented lets encrypt certificate?  I would have thought it would still work without the SSL? 


Edit:
Just to add, it does this no matter if I am inside or outside my local network. 


Edit:
To make this a bit clearer - this installation is not running on an apache server. Apache is running in its own docker - it is just redirecting this traffic to the Docker ports (8080). The PHP seems to correctly initialize but is taking a URL from its location and then getting lost as it tries to initialize.

**Comment by audas** [comment]
Ok it seems it is the same issue as here. 
https://github.com/FreshRSS/FreshRSS/discussions/5968

It looks like FreshRSS can not run in its own docker container on the same domain and can only run on its own subdomain. 

Can I get some indication of where the entry set up is on the PHP and I will take a look at what is happening here as we maybe able to set up an initial pathway like other services like Jellyfin and others.

Thanks.

**Reply by Alkarex** [reply]
It also works on the same domain in a subfolder. You may though face some issues if you have installed FreshRSS at one location and move it to another location. Try to delete your Docker volume to reinstall from a fresh state or edit base_url in `data/config.php`

**Reply by audas** [reply]
How do we edit the config.php through docker? Is there an .ENV file or do we enter the docker via bash ? Can this be set in the Docker run command? I can't see anywhere to set it.

Seems like PHP is setting its location and looking at its headers here.

EDIT:

Looking at the Dockerfile there are indeed EVN variables but nothing to set the base_url.

**Reply by Alkarex** [reply]
The easiest is to redo the install at the right location. 
See the doc https://github.com/FreshRSS/FreshRSS/blob/edge/Docker/README.md#docker-compose
Yes, you can edit `data/config.php` by entering the container or by editing it inside your Docker volume

**Reply by Alkarex** [reply]
P.S. and watch out to clear your browser cache to avoid misunderstandings when you perform manual edits of your location and redirections.
You can also make some tests with curl to be sure
